parcels/examples/example_dask_chunk_OCMs.py

```diff
@@ -0,0 +1,204 @@
+    assert (len(field_set.U.grid.load_chunk) == len(field_set.W.grid.load_chunk))
+    if chunk_mode is False:
+        assert (len(field_set.U.grid.load_chunk) == 1)
+    elif chunk_mode == 'auto':
+        assert (len(field_set.U.grid.load_chunk) != 1)
+    elif chunk_mode == 'specific':
+        assert (len(field_set.U.grid.load_chunk) == (1 * int(math.ceil(201.0/16.0)) * int(math.ceil(151.0/16.0))))
+
+
+# ==== undefined as long as we have no POP example data ==== #
+@pytest.mark.parametrize('mode', ['jit'])  # Only testing jit as scipy is very slow
+@pytest.mark.parametrize('chunk_mode', [False, 'auto', 'specific'])  # Only testing jit as scipy is very slow
+def test_pop(mode, chunk_mode):
+    if chunk_mode == 'auto':
+        dask.config.set({'array.chunk-size': '1MiB'})
+    else:
+        dask.config.set({'array.chunk-size': '128MiB'})
+    field_set = fieldset_from_pop_1arcs(chunk_mode)
+    npart = 20
+    lonp = 70.0 * np.ones(npart)
+    latp = [i for i in -45.0+(-0.25+np.random.rand(npart)*2.0*0.25)]
+    compute_pop_particle_advection(field_set, mode, lonp, latp)
+    # POP sample file dimensions: k=20, j=60, i=60
+    assert (len(field_set.U.grid.load_chunk) == len(field_set.V.grid.load_chunk))
+    assert (len(field_set.U.grid.load_chunk) == len(field_set.W.grid.load_chunk))
+    if chunk_mode is False:
+        assert (len(field_set.U.grid.load_chunk) == 1)
+    elif chunk_mode == 'auto':
+        assert (len(field_set.U.grid.load_chunk) != 1)
+    elif chunk_mode == 'specific':
+        assert (len(field_set.U.grid.load_chunk) == (int(math.ceil(21.0/3.0)) * int(math.ceil(60.0/8.0)) * int(math.ceil(60.0/8.0))))
+
+
+@pytest.mark.parametrize('mode', ['jit'])  # Only testing jit as scipy is very slow
+@pytest.mark.parametrize('chunk_mode', [False, 'auto', 'specific'])  # Only testing jit as scipy is very slow
+def test_globcurrent_2D(mode, chunk_mode):
+    if chunk_mode == 'auto':
+        dask.config.set({'array.chunk-size': '32KiB'})
+    else:
+        dask.config.set({'array.chunk-size': '128MiB'})
+    field_set = fieldset_from_globcurrent(chunk_mode)
+    lonp = [25]
+    latp = [-35]
+    pset = compute_globcurrent_particle_advection(field_set, mode, lonp, latp)
+    # GlobCurrent sample file dimensions: time=UNLIMITED, lat=41, lon=81
+    assert (len(field_set.U.grid.load_chunk) == len(field_set.V.grid.load_chunk))
+    if chunk_mode is False:
+        assert (len(field_set.U.grid.load_chunk) == 1)
+    elif chunk_mode == 'auto':
+        assert (len(field_set.U.grid.load_chunk) != 1)
+    elif chunk_mode == 'specific':
+        assert (len(field_set.U.grid.load_chunk) == (1 * int(math.ceil(41.0/16.0)) * int(math.ceil(81.0/16.0))))
+    assert(abs(pset[0].lon - 23.8) < 1)
+    assert(abs(pset[0].lat - -35.3) < 1)
+
+
+@pytest.mark.parametrize('mode', ['jit'])  # Only testing jit as scipy is very slow
+def test_diff_entry_dimensions_chunks(mode):
```

Comment:
How is this function different from the `test_nemo_3D()` above?

Reply:
Look at the dimensions - here, we test that we do not chunk the depth dimension, even if it's in the chunk dictionary, as long as it is not also listed in the `dimensions` dictionary. Those two do have some affect on one another.